# Assignment:- 2(Estimation & Confidece Intervals)

## 1. Using Sample Standard Deviation

In [100]:
import numpy as np
import scipy.stats as stats
import statistics

In [101]:
#sample data
data=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
data

[1.13,
 1.55,
 1.43,
 0.92,
 1.25,
 1.36,
 1.32,
 0.85,
 1.07,
 1.48,
 1.2,
 1.33,
 1.18,
 1.22,
 1.29]

In [102]:
#length of sample data
n=len(data)  
n

15

In [103]:
mean=np.mean(data)   #sample mean
mean

1.2386666666666666

In [104]:
sd=np.std(data)   #sample standard deviation
sd

0.18661427836285438

In [105]:
df=n-1   #degree of freedom
df

14

In [106]:
# t-critical value for 99% confidence interval

alpha = 0.01
t_critical =stats.t.ppf(1-alpha/2,df)
t_critical

2.976842734370834

In [107]:
# margin of error
margin_error = t_critical * (sd / np.sqrt(n))
margin_error

0.14343499804054372

In [108]:
# confidence interval(lower)

CI_lower_bound = mean - margin_error
CI_lower_bound

1.095231668626123

In [109]:
# # confidence interval(upper)

CI_upper_bound = mean + margin_error
CI_upper_bound

1.3821016647072102

### Interpretation

we use t-distribution because the population standard deviation is unknown and also sample size is small.

using a t-distribution, we are 99% confident  that the true mean number of charecters printed before the print-head fails lies between the 1.095231668626123 and 1.3821016647072102.

that means if we took many similar samples of size 15 and build a 99% confidence interval each time, about 99% of those intervals would contain the true population mean.

## 2.Using Known Population Standard Deviation

In [110]:
from scipy.stats import norm

In [111]:
std=0.2
std

0.2

In [112]:
mean

1.2386666666666666

In [113]:
z_critical = norm.ppf(1 - 0.01 / 2)
z_critical

2.5758293035489004

In [114]:
# margin of error 
margin_error1 = z_critical * ( std / np.sqrt(n))
margin_error1

0.13301525327090588

In [115]:
lower_bound_CI = mean - margin_error1
lower_bound_CI

1.1056514133957607

In [116]:
upper_bound_CI = mean + margin_error1
upper_bound_CI

1.3716819199375725

### Interpretation

we are 99% confident that the true population mean number of characters printed before failure lies between 1.11 million and 1.38 million characters.

if we repeat this process many times , about 99% such intervals would capture the true population mean.

## THE END